src/Symfony/Component/Messenger/Stamp/RedeliveryStamp.php

```diff
@@ -24,12 +24,12 @@ final class RedeliveryStamp implements StampInterface
     private $exceptionMessage;
     private $flattenException;
 
-    public function __construct(int $retryCount, string $exceptionMessage = null, FlattenException $flattenException = null)
+    public function __construct(int $retryCount, string $exceptionMessage = null, FlattenException $flattenException = null, \DateTimeTimmutable $redeliveredAt = null)
     {
         $this->retryCount = $retryCount;
         $this->exceptionMessage = $exceptionMessage;
         $this->flattenException = $flattenException;
-        $this->redeliveredAt = new \DateTimeImmutable();
+        $this->redeliveredAt = $redeliveredAt instanceof \DateTimeImmutable ? $redeliveredAt : new \DateTimeImmutable();
```

Comment:
```suggestion
        $this->redeliveredAt = $redeliveredAt ?? new \DateTimeImmutable();
```

i think you can rewrite like this